src/runtime/security/negotiation_service.h

```diff
@@ -20,22 +20,32 @@
 #include "server_negotiation.h"
 
 #include <dsn/cpp/serverlet.h>
+#include <dsn/tool-api/zlocks.h>
 
 namespace dsn {
 namespace security {
+typedef std::unordered_map<rpc_session *, std::unique_ptr<negotiation>> negotiation_map;
 
 class negotiation_service : public serverlet<negotiation_service>,
                             public utils::singleton<negotiation_service>
 {
 public:
+    static void on_rpc_connected(rpc_session *session);
+    static void on_rpc_disconnected(rpc_session *session);
+
     void open_service();
 
 private:
     negotiation_service();
     void on_negotiation_request(negotiation_rpc rpc);
     friend class utils::singleton<negotiation_service>;
     friend class negotiation_service_test;
+
+    static zrwlock_nr _lock; // [
+    static negotiation_map _negotiations;
+    //]
```

Comment:
Why add '//]' here?

Reply:
`//[` and `//]` indicates that  the _lock is used for protecting the members locates between `[` and `]`